iaso/tests/api/payments/test_payments.py

```diff
@@ -74,8 +76,34 @@ def test_update_status(self):
         self.client.force_authenticate(user=self.user)
         response = self.client.patch(f"/api/payments/{self.payment.id}/", format="json", data={"status": "sent"})
         r = self.assertJSONResponse(response, 200)
-        self.assertEqual(r["status"], m.Payment.Statuses.SENT)
+        self.assertEqual(r["status"], PaymentStatuses.SENT)
         self.assertEqual(r["updated_by"], self.user.id)
         self.payment_lot.refresh_from_db()
         self.assertEqual(self.payment_lot.status, m.PaymentLot.Statuses.SENT)
         self.assertEqual(am.Modification.objects.count(), 2)
+
+    def test_dropdown_options(self):
+        self.client.force_authenticate(user=self.user)
+        response = self.client.get(f"/api/payments/options/")
+        r = self.assertJSONResponse(response, 200)
+        for item in r:
+            if item["value"] == PaymentStatuses.PENDING:
+                self.assertEqual(item["label"], "Pending")
+            elif item["value"] == PaymentStatuses.SENT:
+                self.assertEqual(item["label"], "Sent")
+            elif item["value"] == PaymentStatuses.REJECTED:
+                self.assertEqual(item["label"], "Rejected")
+            elif item["value"] == PaymentStatuses.PAID:
+                self.assertEqual(item["label"], "Paid")
+        # Passingthe header works in the front-end, but fails the test
+        # response = self.client.get(f"/api/payments/options/", headers={"Accept-Language": "fr"})
+        # r = self.assertJSONResponse(response, 200)
+        # for item in r:
+        #     if item['value'] == PaymentStatuses.PENDING:
+        #         self.assertEqual(item['label'], 'En attente')
+        #     elif item['value'] == PaymentStatuses.SENT:
+        #         self.assertEqual(item['label'], 'Envoyé')
+        #     elif item['value'] == PaymentStatuses.REJECTED:
+        #         self.assertEqual(item['label'], 'Rejeté')
+        #     elif item['value'] == PaymentStatuses.PAID:
+        #         self.assertEqual(item['label'], 'Payé')
```

Comment:
This works for me locally.

My guess is that you just need to [reset the active language at the end of the test](https://docs.djangoproject.com/fr/4.2/topics/testing/tools/#setting-the-language) so that other tests don't fail?

One way to do it is:

```
from django.conf import settings
from django.utils import translation


# …

translation.activate(settings.LANGUAGE_CODE)
```

Reply:
I ran into the same issue: tests working locally but failing in the CI.

It turns out `compilemessages` was not run in CI. This will be fixed in https://github.com/BLSQ/iaso/pull/1538